modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/resource/HeadlessBuilderResourceImpl.java

```diff
@@ -130,6 +133,16 @@ private <T> Response _executeEndpoint(
 				"Endpoint /%s does not exist for " + path);
 		}
 
+		if (((endpoint.getPathParameter() == null) ||
+			 Objects.equals(endpoint.getPathParameter(), "")) &&
+			Objects.equals(
+				endpoint.getRetrieveType(),
+				APIApplication.Endpoint.RetrieveType.SINGLE_ELEMENT)) {
+
+			throw new InvalidParameterException(
```

Comment:
I don't understand why we included this validation here and not in the Listener

Reply:
If you remember the Figma's design, a user can not indicate the Path Parameter value, as well as the Response Schema Object, both of them are in the Endpoint Configuration Tab. So if the user makes a request, it would throw an InvalidParameterException.